### PR TITLE
attempt to diagnose and prevent certain shutdown issues

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3657/

File: arangod/Aql/QueryRegistry.cpp [hunk 1]
```diff
@@ -181,9 +201,10 @@ void QueryRegistry::destroy(std::string const& vocbase, QueryId id, int errorCod
           TRI_ERROR_BAD_PARAMETER, "query with given vocbase and id not found");
     }
 
-    if (q->second->_isOpen) {
+    if (q->second->_isOpen && !ignoreOpened) {
       // query in use by another thread/request
       q->second->_query->kill();
+      q->second->_expires = TRI_microtime();
```
Comment (graetzer) [inline arangod/Aql/QueryRegistry.cpp]: why not 0?

File: arangod/RestServer/QueryRegistryFeature.cpp [hunk 2]
```diff
@@ -174,6 +174,12 @@ void QueryRegistryFeature::prepare() {
 
 void QueryRegistryFeature::start() {}
 
+void QueryRegistryFeature::stop() {
+  TRI_ASSERT(_queryRegistry != nullptr);
+  _queryRegistry->disallowInserts();
```
Comment (graetzer) [inline arangod/RestServer/QueryRegistryFeature.cpp]: whats the differrence between calling `disallowInserts` on shutdown and checking `application_features::ApplicationServer::isStopping()` ? 
Seems to me like one or the other would be sufficient

File: arangod/Aql/QueryRegistry.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/Aql/QueryRegistry.cpp]: ok, convinced

File: arangod/RestServer/QueryRegistryFeature.cpp (hunk 2 above)
Comment (jsteemann) [reply arangod/RestServer/QueryRegistryFeature.cpp]: ok, I will unify it.

**Comment by jsteemann** [comment]
Obsolete now

**Pull request closed without merging**